Assets/MRTK/Core/Providers/InputSimulation/BaseInputSimulationService.cs

```diff
@@ -66,66 +67,74 @@ public override IMixedRealityController[] GetActiveControllers()
 
         #endregion BaseInputDeviceManager Implementation
 
-        // Register input sources for hands based on hand data
-        protected void UpdateHandDevice(HandSimulationMode simulationMode, Handedness handedness, SimulatedHandData handData)
+        // Register input sources for controllers based on controller data
+        protected void UpdateControllerDevice(ControllerSimulationMode simulationMode, Handedness handedness, object controllerData)
         {
-            if (handData != null && handData.IsTracked)
-            {
-                SimulatedHand controller = GetOrAddHandDevice(handedness, simulationMode);
-                controller.UpdateState(handData);
-            }
-            else
+            if (controllerData != null)
             {
-                RemoveHandDevice(handedness);
+                if(controllerData is SimulatedHandData handData && handData.IsTracked)
+                {
+                    
+                    SimulatedHand hand = GetOrAddControllerDevice(handedness, simulationMode) as SimulatedHand;
+                    hand.UpdateState(handData);
+                    return;
+                    
+                }
             }
+            
+            RemoveControllerDevice(handedness);
+            
         }
 
-        public SimulatedHand GetHandDevice(Handedness handedness)
+        public BaseController GetControllerDevice(Handedness handedness)
```

Comment:
We probably need a section in the release notes that explain the reason for the breaking changes.

The notes would be added here: https://microsoft.github.io/MixedRealityToolkit-Unity/Documentation/ReleaseNotes.html#breaking-changes

Reply:
Agreed, but if you don't mind I want to add the section in the next PR as things will be clearer there.

Reply:
@CDiaz-MS Just realized that this is actually not a breaking change as the original function with the same signature was moved down to the obsolete section. But I will still mention this in the next PR.